server/models/Team.ts

```diff
@@ -144,6 +145,11 @@ class Team extends ParanoidModel {
     );
   }
 
+  getAllowedDomains = async function (): Promise<string[]> {
+    const teamDomains: TeamDomain[] = await this.getTeamDomains();
+    return (teamDomains || []).map((teamDomain) => teamDomain.name);
+  };
```

Comment:
I wrote this method so that `isDomainAllowed` would work without needing to know if `this.teamDomains` was hydrated already

But after being in the repo for a bit, maybe that isn't the best way? It seemed like it would be easy to control that all the usages DID have `TeamDomain` hydrated already, so I could lean that direction.

Do you have a preference?

Reply:
With the naming change below I think this will do the same thing:

```
await team.$get("allowedDomains");
```